### PR TITLE
Pc 384 park home with logic

### PR DESCRIPTION
**Pull request by xnw9**
- "Park home flow"
- Edit summary park to coop with new flow

Essentially duplicating the normal flow

### CODE REVIEW

File: help_to_heat/frontdoor/eligibility.py [hunk 1]
```diff
@@ -29,6 +29,11 @@ def calculate_eligibility(session_data):
     council_tax_band = session_data.get("council_tax_band")
     country = session_data.get("country")
     benefits = session_data.get("benefits")
+    property_type = session_data.get("property_type")
+
+    # "Scenario 0"
```
Comment (xnw9) [inline help_to_heat/frontdoor/eligibility.py:34]: Presuming all park homes are automatically eligible for GBIS

File: help_to_heat/frontdoor/views.py [hunk 2]
```diff
@@ -149,26 +149,38 @@ def change_page_view(request, session_id, page_name):
     return page_map[page_name](request, session_id, page_name, is_change_page=True)
 
 
-def get_prev_next_page_name(page_name):
-    if page_name in schemas.page_prev_next_map:
-        prev_page_name = schemas.page_prev_next_map[page_name]["prev"]
-        next_page_name = schemas.page_prev_next_map[page_name]["next"]
+def get_prev_next_page_name(page_name, session_id=None):
```
Comment (xnw9) [inline help_to_heat/frontdoor/views.py:152]: Main changes are here; it switches between two flows (?)

Comment (xnw9) [reply help_to_heat/frontdoor/views.py:152]: I think what I wrote here is.. hacky 🤦

File: help_to_heat/frontdoor/views.py [hunk 3]
```diff
@@ -557,20 +572,32 @@ def get_context(self, request, session_id, *args, **kwargs):
...
             }
             for page_name, questions in schemas.household_pages.items()
             for question in questions
-            if question in session_data
-            if question in schemas.summary_map
+            if self.show_question(session_data, question)
         )
         return {"summary_lines": summary_lines}
 
+    def show_question(self, session_data, question):
```
Comment (xnw9) [inline help_to_heat/frontdoor/views.py:583]: Not showing property type summary line for park homes, even if we've got answer for property type

File: help_to_heat/frontdoor/views.py [hunk 4]
```diff
@@ -557,20 +572,32 @@ def get_context(self, request, session_id, *args, **kwargs):
...
+            return question_answered
+
     def get_answer(self, session_data, question):
         answer = session_data.get(question)
         answers_map = schemas.check_your_answers_options_map.get(question)
         return answers_map[answer] if answers_map else answer
 
+    def get_change_url(self, session_id, question, page_name):
+        if question == "park_home":
+            return reverse("frontdoor:page", kwargs=dict(session_id=session_id, page_name=page_name))
```
Comment (xnw9) [inline help_to_heat/frontdoor/views.py:598]: Need to go through the journey again if one attempts to change park home answer from summary page (confirmed with Ladun

File: help_to_heat/frontdoor/eligibility.py (hunk 1 above)
Comment (CheshireSwift) [reply help_to_heat/frontdoor/eligibility.py:34]: Presuming? Or is that the intended/specified behaviour?

File: help_to_heat/frontdoor/views.py (hunk 2 above)
Comment (CheshireSwift) [reply help_to_heat/frontdoor/views.py:152]: I think it's not ideal, but I suspect it will be replaced when we implement PC-498. Do you agree?

**Review by CheshireSwift — COMMENTED** [review]
Couple of queries.

**Comment by xnw9** [comment]
@CheshireSwift Sorry I can't seem to reply your comments 😢 

Looking at the powerpoint, they didn't specify what schemes park homes are eligible for, but by the look of the flow, park homes will automatically be eligible (they don't have EPC etc...). @LadunOmideyiSoftwire please shout if you think we need to confirm with the client etc.!

I hope so! I hope it's not too hacky to make it impossible to implement PC-498 (?) 😅

File: help_to_heat/frontdoor/eligibility.py (hunk 1 above)
Comment (xnw9) [reply help_to_heat/frontdoor/eligibility.py:34]: Looking at the powerpoint, they didn't specify what schemes park homes are eligible for, but by the look of the flow, park homes will automatically be eligible (they don't have EPC etc...). @LadunOmideyiSoftwire  please shout if you think we need to confirm with the client etc.!

File: help_to_heat/frontdoor/views.py (hunk 2 above)
Comment (xnw9) [reply help_to_heat/frontdoor/views.py:152]: I hope so! I hope it's not too hacky to make it impossible to implement PC-498 (?) 😅

**Pull request merged**